internal/rest/resources/resources.go

```diff
@@ -46,3 +49,47 @@ var ExtendedEndpoints = rest.Resources{
 	Path:      rest.EndpointType(client.ExtendedEndpoint),
 	Endpoints: []rest.Endpoint{},
 }
+
+// checkDuplicateEndpoints checks if any endpoints defined in extensionServers conflict with internal endpoints.
+func checkDuplicateEndpoints(extensionServerEndpoints rest.Resources, extendedEndpoints rest.Resources) error {
+	allExistingEndpoints := []rest.Resources{UnixEndpoints, PublicEndpoints, InternalEndpoints, extendedEndpoints}
+	existingEndpointPaths := make(map[string]bool)
+
+	for _, endpoints := range allExistingEndpoints {
+		for _, e := range endpoints.Endpoints {
+			url := filepath.Join(string(endpoints.Path), e.Path)
+			existingEndpointPaths[url] = true
+		}
+	}
+
+	for _, e := range extensionServerEndpoints.Endpoints {
+		url := filepath.Join(string(extensionServerEndpoints.Path), e.Path)
+		if existingEndpointPaths[url] {
+			return fmt.Errorf("Endpoint %q conflicts with internal endpoint", url)
+		}
+	}
+
+	return nil
+}
+
+// MergeExtendedEndpoints merges the endpoints from extensionServers with endpoints defined in extendedEndpoints.
+// It also ensures that there are no conflicts between endpoints from extensionServers and internal endpoints.
+func MergeExtendedEndpoints(extensionServers []rest.Server, extendedEndpoints rest.Resources) ([]rest.Resources, error) {
```

Comment:
Is there a better place to keep `checkDuplicateEndpoints` and `MergeExtendedEndpoints`?

Reply:
I think there should be fine. It's under `internal` so that's really the biggest concern :)